pkg/lib/cockpit-components-table.scss

```diff
@@ -1,14 +1,90 @@
 @import "~@patternfly/patternfly/components/Table/table.scss";
 @import "~@patternfly/patternfly/components/Table/table-grid.scss";
 
-// Display the actions next to the header inline
-header {
-    display: flex;
-    > :only-child {
+.ct-table {
+    &.pf-m-compact {
+        > thead, > tbody {
+            > tr:not(.pf-c-table__expandable-row) {
+                // We actually want the normal font size for our lists
+                --pf-c-table-cell--FontSize: var(--pf-global--FontSize--md);
+            }
+        }
+    }
+
+    &-header {
+        display: flex;
+        align-items: center;
+        flex-wrap: wrap;
+
+        > :only-child {
+            flex: auto;
+        }
+    }
+
+    &-heading {
+        // Push buttons to the right by stretching the heading
         flex: auto;
+        // Add a bit of minimum margin to the right of the heading
+        margin-right: var(--pf-global--spacer--md);
+        // Set a minimum height of 3rem, so when buttons wrap, there's spacing
+        min-height: var(--pf-global--spacer--2xl);
+        // Make sure textual content is aligned to the center
+        display: flex;
+        align-items: center;
     }
-}
 
-thead.listing-ct-empty tr td {
-    font-weight: normal;
-}
+    &-actions {
+        > * {
+            margin: var(--pf-global--spacer--xs) 0 var(--pf-global--spacer--xs) var(--pf-global--spacer--sm);
+        }
+
+        > :first-child {
+            margin-left: 0;
+        }
+    }
+
+    &-empty {
+        tr td {
+            // FIXME: Not sure if this is needed. Was a direct port.
+            font-weight: normal;
+        }
+    }
+
+    // Fix toggle button alignment
+    .pf-c-table__toggle {
+        // Workaround: Chrome sometimes oddly expands the table,
+        // unless a width is set. (This affects panels the most, but not only.)
+        // As the width is smaller than the contents, and this is a table,
+        // the cell will stay at the correct width.
+        width: 1px;
+
+        > .pf-c-button {
+            padding-top: var(--pf-global--spacer--sm);
+            padding-bottom: var(--pf-global--spacer--sm);
+            vertical-align: top;
+            // Not every browser supports text-top, so define top (above) as a fallback
+            vertical-align: text-top;
+        }
+    }
+
+    // Shrink buttons down a little
```

Comment:
Why do we want to shrink the buttons  here?

Reply:
Alignment is completely whacked with the normal sized buttons.

I can try to accommodate it though. I don't remember if this was before or after I undid the font-size shrinking due to `pf-m-compact`.